openstates/ri/legislators.py

```diff
@@ -25,123 +31,72 @@ def scrape(self, chamber, term):
+        for row in contact_page.xpath('//tr[@valign="TOP"]'):
+            tds = row.xpath('td')
+            (detail_link, ) = tds[link_col_ix].xpath('.//a/@href')
+            # Ignore name (2nd col). We have a regex built up below for the spreadsheet name I don't want to touch
+            district, _, email, phone = [td.text_content().strip() for td in tds[:link_col_ix]]
+            contact_info_by_district[district] = {
+                'email': email,
+                'phone': phone,
+                'detail_link': detail_link,
+            }
+
         self.urlretrieve(url, 'ri_leg.xls')
 
         wb = xlrd.open_workbook('ri_leg.xls')
         sh = wb.sheet_by_index(0)
 
-        # This isn't perfect but it's cheap and better than using the
-        # XLS doc as the source URL for all legislators.
-        # 374: RI: legislator url
-        leg_source_url_map = {}
-        leg_page = self.lxmlize(source_url)
-
-        for link in leg_page.xpath('//td[@class="ms-vb2"]'):
-            leg_name = link.text_content().replace(source_url_title_replacement,'')
-            leg_url = link.xpath("..//a")[0].attrib['href']
-            leg_source_url_map[leg_name] = leg_url
-
         for rownum in xrange(1, sh.nrows):
-            d = {}
-            for field, col_num in excel_mapping.iteritems():
-                d[field] = sh.cell(rownum, col_num).value
+            d = {
+                field: sh.cell(rownum, col_num).value
+                for field, col_num in excel_mapping.iteritems()
+            }
 
+            # Convert float to an int, and then to string, the format required by billy
+            district = str(int(d['district']))
             if d['full_name'].upper() == "VACANT":
                 self.warning(
-                    "District {}'s seat is vacant".format(int(d['district'])))
+                    "District {}'s seat is vacant".format(district))
                 continue
 
-            slug = re.match(
-                "(?P<class>sen|rep)-(?P<slug>.*)@(rilin\.state\.ri\.us|rilegislature\.gov)", d['email']
-            )
-            
-            if 'asp' in d['email']:
-                d['email'] = None
-
-            if d['email'] is not None:
-                info = slug.groupdict()
-                info['chamber'] = "senators" if info['class'] == 'sen' else "representatives"
-
-                url = ("http://www.rilin.state.ri.us/{chamber}/"
-                       "{slug}/Pages/Biography.aspx".format(**info))
-
-            dist = str(int(d['district']))
-            district_name = dist
-
-            assert d['full_name'].startswith(rep_type), "Improper name found"
-            full_name = re.sub(r"^{}(?=\s?[A-Z].*$)".format(rep_type), '', d['full_name']).strip()
-            translate = {
-                "Democrat"    : "Democratic",
-                "Republican"  : "Republican",
-                "Independent" : "Independent"
-            }
-
-            homepage_url = None
```

Comment:
Modified to the portion that generates `contact_info_by_district` at the top of this method.